### ISSUE TITLE
Wrong regex for reserved nodes

### ISSUE THREAD

**Issue by matejzero** [vpenso/prometheus-slurm-exporter#4]
We are running slurm 17.11.2 and exporter is not reporting reserved nodes correctly because of a different output of `sinfo` command.

Exporter is using `%T` field, which prints long format. It should be `%t`.

From [slurm](https://slurm.schedmd.com/sinfo.html) manuals:
```
%t
State of nodes, compact form
%T
State of nodes, extended form
```

**Comment by mtds** [comment]
This exporter was developed and tested with Slurm 14.03.9 but we are also using it right
now with a small cluster running Slurm 17.11.3.

At the moment we don't have nodes in *resv* state but do you see a different output from
*sinfo*? I have shortly tested it with both our installation (the cluster in production and the
test one) but the output looks basically the same to me, independently when I am using 
``%t`` or ``%T``:

```
lxhost01,down*  
lxhost02,down*  
lxhost03,idle  
lxsubmit01,idle 
```

Which output do you see on your cluster with the ``%T`` format string?

**Comment by matejzero** [comment]
Thanks for getting back at me.

With %T:
```
# sinfo -h -o %n,%T
wn124,down*
wn139,draining
wn062,reserved
wn012,mixed
wn011,allocated
```

With %t
```
# sinfo -h -o %n,%t
wn124,down*
wn139,drng
wn061,resv
wn062,resv
wn125,mix
wn011,alloc
```

**Comment by mtds** [comment]
Ok, I see what you mean. Before committing the change in the repo I just want 
to double check on our cluster but in principle correcting the problem should be 
easy, even without changing the ``sinfo`` command used by the exporter.

You can check the regex used by ``nodes.go`` at line 80:

``resv := regexp.MustCompile(`^resv`) ``

It could be also in this form:

``resv := regexp.MustCompile(`^res`) ``

In this case we will catch all the nodes in ``reserved`` state, independently of the
output modifier option. Could you try to modify the code on your side and compiling 
it with the GOlang compiler? If it is working correctly I will commit the modification.

**Comment by mtds** [comment]
Ok, I have committed the fix (``02d874b``).

Thanks for spotting this, we have definitely missed it!

**Comment by matejzero** [comment]
Great!

I recompiled it and now it shows the correct node count. Thanks for the fix.
